### DISCUSSION TITLE
How to use markdown (bold, italics ...) in messages

### DISCUSSION THREAD

**Discussion by bublath** [AsamK/signal-cli#878] [Q&A]
I already have searched a bit and found a discussion to support markdown like *bold* _underline_ 'code' ... in Signal, but if I get it right, even after years of discussion, no conclusion was reached and it's not implemented.
While it's possible to send pre-formatted unicode text (copy&paste), especially when using signal-cli (from a console or script) it would be helpful to have a way to use some kind of tags in plain ASCII. An option could also be to use html tags like <b>bold</b>.
Is there already a way to do it, and I'm just not finding it?
Whatsapp and Telegram have simple ways to do this and I have users asking for this in order to switch to Signal.

Edit - github turned all my examples in real formatting, so I'm taking about using asterisks for bold, underscore for underline, tics for code and the html tag "b" for bold

**Comment by AsamK** [comment]
As far as I know it's not possible.
Also in the official Android client I haven't seen any code or protocol extensions that would support formatting.

**Reply by bublath** [reply]
Is there are good library that could be integrated to support something like this? Something like markdown->unicode or html->unicode since providing pre-formatted utf-8 that has formatting works well.
I already searched for a Perl lib, so I could do a pre-processing before sending, but didn't find any.

**Reply by Miniontoby** [reply]
The app has been updated since less than a week now. 

The commit is here: https://github.com/signalapp/Signal-Android/commit/cc490f4b731eb6675e27439829dcfd8c1a68ce13

So maybe it can now be added?

**Reply by AsamK** [reply]
It's already implemented now, with --text-style
https://github.com/AsamK/signal-cli/blob/master/man/signal-cli.1.adoc#send

**Reply by Miniontoby** [reply]
Yeah, but not for dbus...

Btw is there a way to get the text style of a message?

**Reply by Miniontoby** [reply]
> It's already implemented now, with --text-style https://github.com/AsamK/signal-cli/blob/master/man/signal-cli.1.adoc#send

But how to use multiple text styles?

**Reply by exquo** [reply]
> Yeah, but not for dbus...

Opened an issue: https://github.com/AsamK/signal-cli/issues/1271.
